### PR TITLE
add feature gate for scheduler and use 1 place for default feature gates

### PR DESCRIPTION
**Pull request by KodieGlosserIBM**
Related https://github.com/openshift/hypershift-toolkit/issues/89

### CODE REVIEW

File: assets/kube-scheduler/kube-scheduler-deployment.yaml [hunk 1]
```diff
@@ -55,6 +55,9 @@ spec:
         - "--authentication-kubeconfig=/etc/kubernetes/secret/kubeconfig"
         - "--authorization-kubeconfig=/etc/kubernetes/secret/kubeconfig"
         - "-v=2"
+{{if .ExtraFeatureGates }}
+        - "--feature-gates={{ range $featureGate := .ExtraFeatureGates }}{{ $featureGate }},{{ end }}"
```
Comment (relyt0925) [inline assets/kube-scheduler/kube-scheduler-deployment.yaml]: pass them as a feature gate per line so you don't have to worry about a trailing comma situation

Comment (relyt0925) [reply assets/kube-scheduler/kube-scheduler-deployment.yaml]: they are additive when you pass them one at a time like
-- feature-gates={{featuregate1}}
-- feature-gates={{featuregate2}}
-- feature-gates={{featuregate3}}

File: cluster.yaml.example [hunk 2]
```diff
@@ -143,3 +143,4 @@ openVPNServerResources:
           memory: 200Mi
 apiServerAuditEnabled: true
 restartDate: "2017-07-27T16:39:00Z"
+extraFeatureGates: [ExpandInUsePersistentVolumes=true,Accelerators=true,ExperimentalCriticalPodAnnotation=true]
```
Comment (derekwaynecarr) [inline cluster.yaml.example]: What is Accelerators?

I see no corresponding feature:
https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go

Keep in mind that OpenShift tightly controls what gates are enabled or disabled and we restrict upgradeability around the use of particular FeatureSets.

See:
https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L99

I would expect ROKS to use the same FeatureSet as  our default features list (both enabled and disabled):
https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L113

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
Can you clarify the actual list of feature gates being supplied?

File: cluster.yaml.example (hunk 2 above)
Comment (KodieGlosserIBM) [reply cluster.yaml.example]: @derekwaynecarr we use `ExpandInUsePersistentVolumes=true` but the others I added just for an example. I can remove the others.

File: assets/kube-scheduler/kube-scheduler-deployment.yaml (hunk 1 above)
Comment (KodieGlosserIBM) [reply assets/kube-scheduler/kube-scheduler-deployment.yaml]: done

File: assets/kube-scheduler/kube-scheduler-deployment.yaml [hunk 3]
```diff
@@ -55,6 +55,9 @@ spec:
         - "--authentication-kubeconfig=/etc/kubernetes/secret/kubeconfig"
         - "--authorization-kubeconfig=/etc/kubernetes/secret/kubeconfig"
         - "-v=2"
+{{if .ExtraFeatureGates }}{{ range $featureGate := .ExtraFeatureGates }}
+        - "--feature-gates={{ $featureGate }}"
```
Comment (csrwng) [inline assets/kube-scheduler/kube-scheduler-deployment.yaml:62]: should we not statically include feature gates that are not in .ExtraFeatureGates?
https://github.com/openshift/hypershift-toolkit/blob/master/assets/kube-apiserver/config.yaml#L28-L30

Comment (csrwng) [reply assets/kube-scheduler/kube-scheduler-deployment.yaml:62]: Even better, specify them all in the config and just use the ones from the config everywhere, that way we make sure they're consistent across components

Comment (KodieGlosserIBM) [reply assets/kube-scheduler/kube-scheduler-deployment.yaml:62]: done

**Comment by KodieGlosserIBM** [comment]
@derekwaynecarr can you please review and let me know if you have any concerns?

**Pull request merged**